docs/concepts/cluster-administration/cloud-providers.md

```diff
@@ -58,50 +58,108 @@ The information for the annotations for AWS is taken from the comments on [aws.g
+  be added in the future.
+‡ Load Balancing V1 API support is deprecated.
+
+Service discovery is achieved by listing the service catalog managed by
+OpenStack Identity (Keystone) using the `auth-url` provided in the provider
+configuration. The provider will gracefully degrade in functionality when
+OpenStack services other than Keystone are not available and simply disclaim
+support for impacted features. Certain features are also enabled or disabled
+based on the list of extensions published by Neutron in the underlying cloud.
 
 ## cloud.conf
 Kubernetes knows how to interact with OpenStack via the file cloud.conf. It is the file that will provide Kubernetes with credentials and location for the OpenStack auth endpoint.
 You can create a cloud.conf file by specifying the following details in it
 
-### Minimal configuration
-This is an example of a minimal configuration that touches the values that most often need to be set:
+### Typical configuration
+This is an example of a typical configuration that touches the values that most
+often need to be set. It points the provider at the OpenStack cloud's Keystone
+endpoint, provides details for how to authenticate with it, and configures the
+load balancer:
 
 ```yaml
 [Global]
-username=user  
-password=pass  
-auth-url=https://<keystone_ip>/identity/v3  
-tenant-id=c869168a828847f39f7f06edd7305637  
+username=user
+password=pass
+auth-url=https://<keystone_ip>/identity/v3
+tenant-id=c869168a828847f39f7f06edd7305637
 domain-id=2a73b8f597c04551a0fdc8e95544be8a
 
 [LoadBalancer]
-subnet-id=6937f8fa-858d-4bc9-a3a5-18d2c957166a  
+subnet-id=6937f8fa-858d-4bc9-a3a5-18d2c957166a
 ```
 
 #### Global
-* `username`: Refers to the username of a valid user set in keystone.
-* `password`:Refers to the password of a valid user set in keystone.
-* `auth-url`: The URL of the keystone API used to authenticate. On OpenStack control panels, this can be found at Access and Security > API Access > Credentials.
-* `tenant-id`: Used to specify the id of the project where you want to create your resources.
-* `domain-id`: Used to specify the id of the domain your user belongs to.
+These configuration options for the OpenStack provider pertain to its global
+configuration and should appear in the `[Global]` section of the `cloud.conf`
+file:
 
-####  Load Balancer
-* `subnet-id`: Used to specify the id of the subnet you want to create your loadbalancer on. Can be found at Network > Networks. Click on the respective network to get its subnets.
+* `username` (Required): Refers to the username of a valid user set in keystone.
+* `password` (Required): Refers to the password of a valid user set in keystone.
+* `auth-url` (Required): The URL of the keystone API used to authenticate. On OpenStack control panels, this can be found at Access and Security > API Access > Credentials.
+* `tenant-id` (Required): Used to specify the id of the project where you want to create your resources.
```

Comment:
Should add a comment about how Keystone v3 has changes tenant to project, and in the future this terminology will be changing.

Reply:
Adding a mention, though not clear we got consensus on changing the K8S facing aspect from the discussion in https://github.com/kubernetes/kubernetes/pull/52595 - for now it just maps automatically.